### PR TITLE
DOP-929: Add drivers-index template functionality

### PR DESCRIPTION
**Pull request by mango-db**
### Stories/Links:

[DOP-929](https://jira.mongodb.org/browse/DOP-929)
[merged parser PR](https://github.com/mongodb/snooty-parser/pull/348)
[merged docs-ecosystem PR](https://github.com/mongodb/docs-ecosystem/pull/766)

🟢 ALL SYSTEMS GO FOR LAUNCH 🟢 

### Staging Links:

[docs-ecosystem](https://docs-mongodborg-staging.corp.mongodb.com/DOP-929/drivers/cgoecknerwald/DOP-929/)
[docs-realm](https://docs-mongodborg-staging.corp.mongodb.com/master/drivers/cgoecknerwald/DOP-929/) -> adding because I made changes to the `document` template + `DocumentBody` component
[docs](https://docs-mongodborg-staging.corp.mongodb.com/master/docs/cgoecknerwald/DOP-929/index/) -> adding because I removed the `height` and `width` args from the SVG Icons, which are used in the language dropdown selectors + tab selectors.

### Notes:
- Removed use of `project` and `slug` keywords in `getTemplate` function
- Removed use of `StaticQuery` in `DocumentBody.js`, as `getTemplate` function no longer needs `project` arg
- Changed `PropType` of `parentPaths` from `arrayOf.(string)` to `object`, as it was complaining @rayangler 
- I somewhat randomly chose to use the terminology "index" instead of "landing" as a replacement for "homepage"... could change to "landing" if people feel very strongly about it?
- I removed the height and width values from the SVGs, as they are unnecessary and were interfering
- Upon refresh, the border shadows around the tiles flashes and disappears. I believe this is because of the non-Card-standard styling we are applying. This is happening on the live site. I am erring in favor of not making design changes.
- I rewrote the CSS for `DriverIndexTiles` to make it more robust + kill a bug that caused this:
Screen Shot 2021-10-06 at 5 47 41 PM

### CODE REVIEW

File: src/templates/document.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ Document.propTypes = {
       children: PropTypes.array,
       options: PropTypes.object,
     }).isRequired,
-    parentPaths: PropTypes.arrayOf(PropTypes.string),
+    parentPaths: PropTypes.object,
```
Comment (rayangler) [inline src/templates/document.js:75]: Thanks for catching this. I totally forgot to change it when I last touched breadcrumbs

**Review by rayangler — COMMENTED** [review]
Code LGTM for now (will take another look tomorrow/after merge conflicts). `DriversIndex` sounds good to me. 

Also, is the drivers staging link up to date?

File: src/layouts/index.js [hunk 2]
```diff
@@ -41,7 +40,7 @@ const globalCSS = css`
     width: 0;
   }
 
-  // Originally from docs-tools navbar.css
+  ${'' /* Originally from docs-tools navbar.css */}
```
[the remark this answers is not in the document]
Comment (mango-db) [reply src/layouts/index.js:43]: Honestly, me too

**Comment by mango-db** [comment]
I need to rewrite the CSS in the `document` template to handle mobile, so this PR is unofficially "draft" again

File: src/layouts/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mango-db) [reply src/layouts/index.js:43]: Oh yeah, it's because it wasn't applying comment color correctly and I got annoyed

**Comment by mango-db** [comment]
We're good to go 🚗

File: src/components/DriversIndexTiles.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    title: 'Swift',
+    icon: <IconSwift />,
+  },
+];
+
+const StyledGrid = styled('div')`
+  display: grid;
+  grid-gap: ${theme.size.default};
+  grid-template-columns: repeat(3, 1fr);
+  padding-right: ${theme.size.xlarge};
```
Comment (rayangler) [inline src/components/DriversIndexTiles.js:91]: (Nit) Is this right padding needed for every screen size? It looks fine to me without it

File: src/styles/ecosystem-homepage.module.css [hunk 4]
```diff
@@ -1,8 +0,0 @@
-.fullWidth {
-  margin: 0px auto !important;
-  padding: 40px 0px;
```
Comment (rayangler) [inline src/styles/ecosystem-homepage.module.css:3]: The page looks pretty different without the 40px between the top nav and where the breadcrumbs start. Just wanted to make sure that this difference is intentional and/or okay!

File: src/components/DriversIndexTiles.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+import IconPHP from './icons/Php';
+import IconPython from './icons/Python';
+import IconRuby from './icons/Ruby';
+import IconRust from './icons/Rust';
+import IconScala from './icons/Scala';
+import IconSwift from './icons/Swift';
+
+// DriversIndexTiles is used to display the drivers as Cards on docs.mongodb.com/drivers
+
+// TODO: Unhardcode this. Ideally, the SVG resources would just be plain SVG, not components
```
[the remark this answers is not in the document]
Comment (rayangler) [reply src/components/DriversIndexTiles.js:23]: Curious: is there an advantage/benefit to having them as SVG files vs. wrapping them as a component?

[the remark this answers is not in the document]
Comment (mango-db) [reply src/components/DriversIndexTiles.js:23]: If they were SVG files, I could pass them as a `src` arg to the `img` tag, which allows easy handling IMO! I don't really see an advantage to wrapping them in a component 🤔

File: src/styles/ecosystem-homepage.module.css (hunk 4 above)
Comment (mango-db) [reply src/styles/ecosystem-homepage.module.css:3]: Difference was unintentional - correcting!

**Review by rayangler — APPROVED** [review]
LGTM! 

The other staging links for docs and realm no longer work but I verified on an older (by maybe 1 or 2 commits) [staging link](https://docs-mongodbcom-staging.corp.mongodb.com/master/docs/raymundrodriguez/DOP-929/indexes/) that the changes related to the templates and SVGs seem fine.

**Pull request merged**